File: sample/src/main/java/com/sunmi/samples/printerx/CashFragment.java
package com.sunmi.samples.printerx;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.sunmi.printerx.SdkException;
import com.sunmi.samples.printerx.databinding.FragmentCashBinding;


public class CashFragment extends Fragment {

    private FragmentCashBinding binding;

    public CashFragment() {
        super();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentCashBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding.cashSwitch.setOnClickListener(view1 -> {
            try {
                ((MainActivity)getActivity()).selectPrinter.cashDrawerApi().open(null);
            } catch (SdkException e) {
                e.printStackTrace();
            }
        });

        binding.cashStatus.setOnClickListener(view12 -> {
            try {
                boolean ret = ((MainActivity)getActivity()).selectPrinter.cashDrawerApi().isOpen();
                if(ret) {
                    binding.cashStatusShow.setText("on");
                } else {
                    binding.cashStatusShow.setText("off");
                }
            } catch (SdkException e) {
                //若接口不支持，可以认为设备不支持钱箱
                binding.cashStatusShow.setText("none");
            }
        });
    }
}

File: sample-ktx/src/main/java/com/sunmi/samples/printerx/ui/main/CashFragment.kt
package com.sunmi.samples.printerx.ui.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import com.sunmi.samples.printerx.R
import com.sunmi.samples.printerx.databinding.FragmentCashBinding
import com.sunmi.samples.printerx.ui.vm.CashViewModel

/**
 *   钱箱使用样例
 *   钱箱仅指具有钱箱接口的商米设备
 */
class CashFragment: Fragment() {

    private lateinit var dataBinding : FragmentCashBinding

    companion object {
        fun newInstance() = CashFragment()
    }


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        dataBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_cash, container, false)
        return dataBinding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val viewModel: CashViewModel by viewModels()
        dataBinding.model = viewModel
        dataBinding.lifecycleOwner = viewLifecycleOwner
        viewModel.status()
    }
}